zeg/config.py

```diff
@@ -42,4 +42,11 @@ def validate_config(configuration):
     with open(schema_path, 'r') as stream:
         schema = yaml.load(stream, Loader=yaml.SafeLoader)
 
-    validate(configuration, schema)
+    try:
+        jsonschema.validate(configuration, schema)
+    except jsonschema.exceptions.ValidationError as ex:
+        log.error(
+            'Configuration validation error. {err}',
+            err=ex.message
+        )
+        sys.exit(1)
```

Comment:
Don't call `sys.exit()` here - either reraise or wrap the `ValidationError` and catch and turn into a return code as a new exception block in `main()`.

Reply:
Why catch in `main`, there are `sys.exit` calls scattered throughout the files. Dont doubt its the right thing to do, just wondering what difference it makes.

Reply:
In short, because it's less code total, and makes it easier for callers to behave in a context-sensitive manner - the testing is simpler for instance.